src/mbed_tools/project/_internal/git_utils.py

```diff
@@ -0,0 +1,97 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Wrappers for git operations."""
+from dataclasses import dataclass
+from pathlib import Path
+
+import git
+
+from mbed_tools.project.exceptions import VersionControlError
+from mbed_tools.project._internal.progress import ProgressReporter
+
+
+@dataclass
+class GitReference:
+    """Git reference for a remote repository.
+
+    Attributes:
+        repo_url: URL of the git repository.
+        ref: The reference commit sha, tag or branch.
+    """
+
```

Comment:
Does black not want to remove this newline? I see in `clone()` below we don't have a newline after the docstring.

Reply:
No, that's expected. According to `Pep257`, in class definitions a newline is required, in function definitions it's not. https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings - `black` formats this way.

Reply:
OK, thanks